pkg/util/fileutil/file.go

```diff
@@ -82,3 +99,39 @@ func RootDir() (string, error) {
 func Base(dir string) string {
 	return filepath.Base(dir)
 }
+
+// ConfigFile returns the location for the specified envd config file
+func ConfigFile(filename string) (string, error) {
+	if strings.ContainsRune(filename, os.PathSeparator) {
+		return "", fmt.Errorf("filename %s should not contain any path separator", filename)
+	}
+	exist, err := DirExists(DefaultConfigDir)
+	if err != nil {
+		return "", err
+	}
+	if !exist {
+		err = os.Mkdir(DefaultConfigDir, os.ModeDir|0700)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to create the config dir")
+		}
+	}
+	return path.Join(DefaultConfigDir, filename), nil
+}
+
+// CacheFile returns the location for the specified envd cache file
+func CacheFile(filename string) (string, error) {
+	if strings.ContainsRune(filename, os.PathSeparator) {
+		return "", fmt.Errorf("filename %s should not contain any path separator", filename)
+	}
+	exist, err := DirExists(DefaultCacheDir)
+	if err != nil {
+		return "", err
+	}
+	if !exist {
+		err = os.Mkdir(DefaultCacheDir, os.ModeDir|0700)
```

Comment:
align with the implementation of "adrg/xdg"

* https://github.com/adrg/xdg/blob/31e6a328d142d3b8ed43fda69c4f2b55e77d8e79/internal/pathutil/pathutil.go#L42-L60